com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -577,9 +586,12 @@ internal void ClientCollectSoftSyncSceneObjectSweep(NetworkObject[] networkObjec
 
             for (int i = 0; i < networkObjects.Length; i++)
             {
-                if (networkObjects[i].IsSceneObject == null)
+                if (networkObjects[i].NetworkManager == NetworkManager)
```

Comment:
overall, it all makes sense.
I only wonder if we should follow the same logic (skip if `NetworkManager` is not mine) in any other places, have you ever scanned through other places to see if we're covering all of them potentially?

Reply:
Almost all places use the owner NetworkManager by default. 

This is the only place we use FindObjectsOfType and thus aren't filtered yet properly.